### ISSUE TITLE
Fix memory leak in SSLSocket.getpeercert()

### ISSUE THREAD

**Issue by b679055b-4f0a-4208-a5af-363e1e65268d** [python/cpython#73924]
BPO | [29738](https://bugs.python.org/issue29738)
--- | :---
Nosy | @tiran, @Mariatta, @olivielpeau
PRs | <li>python/cpython#526</li><li>python/cpython#1142</li><li>python/cpython#1143</li><li>python/cpython#1144</li><li>python/cpython#1146</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/tiran'
closed_at = <Date 2017-04-15.01:39:44.849>
created_at = <Date 2017-03-06.20:10:11.750>
labels = ['expert-SSL', '3.7', 'performance']
title = 'Fix memory leak in SSLSocket.getpeercert()'
updated_at = <Date 2017-04-17.15:43:55.536>
user = 'https://github.com/olivielpeau'
```

bugs.python.org fields:
```python
activity = <Date 2017-04-17.15:43:55.536>
actor = 'olivielpeau'
assignee = 'christian.heimes'
closed = True
closed_date = <Date 2017-04-15.01:39:44.849>
closer = 'Mariatta'
components = ['SSL']
creation = <Date 2017-03-06.20:10:11.750>
creator = 'olivielpeau'
dependencies = []
files = []
hgrepos = []
issue_num = 29738
keywords = []
message_count = 9.0
messages = ['289130', '290101', '291685', '291687', '291688', '291689', '291690', '291691', '291799']
nosy_count = 3.0
nosy_names = ['christian.heimes', 'Mariatta', 'olivielpeau']
pr_nums = ['526', '1142', '1143', '1144', '1146']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'resource usage'
url = 'https://bugs.python.org/issue29738'
versions = ['Python 2.7', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by b679055b-4f0a-4208-a5af-363e1e65268d** [comment]
The code snippet in bpo-25569 reproduces the memory leak with Python 3.6.0 and 2.7.13. The current memory leak is a regression that was introduced in bpo-26470.

Going to attach a PR on github that fixes the issue shortly.

**Comment by b679055b-4f0a-4208-a5af-363e1e65268d** [comment]
I've attached the PR on Github and signed the CLA, is there anything else needed from me to get this reviewed? Thanks!

**Comment by Mariatta** [comment]
New changeset 2849cc34a8db93d448a62d69c462402347b50dcb by Mariatta (Olivier Vielpeau) in branch 'master':
bpo-29738: Fix memory leak in \_get_crl_dp (GH-526)
https://github.com/python/cpython/commit/2849cc34a8db93d448a62d69c462402347b50dcb

**Comment by Mariatta** [comment]
New changeset b2b00e039ce71e69148e5f479e2154c1c7b712ca by Mariatta in branch '2.7':
[2.7] bpo-29738: Fix memory leak in \_get_crl_dp (GH-526) (GH-1144)
https://github.com/python/cpython/commit/b2b00e039ce71e69148e5f479e2154c1c7b712ca

**Comment by Mariatta** [comment]
New changeset 39f5956ffbcc4fe83cee59eed1d18845a5b25dd2 by Mariatta in branch 'master':
bpo-29738: Add Olivier Vielpeau to [Misc/ACKS](https://github.com/python/cpython/blob/main/Misc/ACKS) (GH-1146)
https://github.com/python/cpython/commit/39f5956ffbcc4fe83cee59eed1d18845a5b25dd2

**Comment by Mariatta** [comment]
New changeset 8e7201342dc6eef201bfa4f533ad89a8144fd693 by Mariatta in branch '3.6':
[3.6] bpo-29738: Fix memory leak in \_get_crl_dp (GH-526) (GH-1142)
https://github.com/python/cpython/commit/8e7201342dc6eef201bfa4f533ad89a8144fd693

**Comment by Mariatta** [comment]
New changeset c40740cf06470a774e98f84537801cd8b7176938 by Mariatta in branch '3.5':
[3.5] bpo-29738: Fix memory leak in \_get_crl_dp (GH-526) (GH-1143)
https://github.com/python/cpython/commit/c40740cf06470a774e98f84537801cd8b7176938

**Comment by Mariatta** [comment]
I merged Olivier's PR, and backported into 3.6, 3.5, and 2.7.
I also added Olivier Vielpeau to misc/ACKS.

Thanks :)

**Comment by b679055b-4f0a-4208-a5af-363e1e65268d** [comment]
Thnaks for the reviews and the merge! :)
